Public/Src/Tools/Execution.Analyzer/Args.cs

```diff
@@ -141,9 +141,9 @@ public Args(string[] args)
                 throw Error("Additional executionLog to compare parameter is required");
             }
 
-            // The fingerprint store based cache miss analyzer only uses graph information from the newer build,
-            // so skip loading the graph for the earlier build
-            if (m_mode.Value != AnalysisMode.CacheMiss)
+            // The fingerprint store based cache miss analyzer and the bxl invocation analyzer
+            // only use graph information from the newer build, so skip loading the graph for the earlier build

```

Comment:
The CacheMiss analyzer is a little tricky since it takes two /xl parameter for the first and second build so it can infer the location of the two fingerprint stores. But as the comment states it only consumes the graph from the second build which is why it skips the first builds' XLG here.

I wouldn't expect this to be applicable to your new analyzer since it should only be consuming a single /xl parameter corresponding to the current build.

Reply:
Talked offline -> once more analyzers hit this case, we will switch to using a list or enum instead to hold all modes that don't require a graph to be read in

---
In reply to: [306094308](https://github.com/microsoft/BuildXL/pull/636#discussion_r306094308) [](ancestors = 306094308)